File: analyze_data.js
const fs = require('fs');

const destinations = JSON.parse(fs.readFileSync('D:/Travel-Weather-Website/data/destinations.json', 'utf8'));
const districts = JSON.parse(fs.readFileSync('D:/Travel-Weather-Website/data/districts.json', 'utf8'));

// Analyze existing destinations by district
const existingByDistrict = {};
destinations.forEach(d => {
    const dist = d.district;
    if (!existingByDistrict[dist]) existingByDistrict[dist] = [];
    existingByDistrict[dist].push(d.id);
});

console.log('=== Existing destinations by district ===');
Object.keys(existingByDistrict).sort().forEach(dist => {
    const ids = existingByDistrict[dist];
    console.log(`${dist}: ${ids.length} - ${ids.join(', ')}`);
});

console.log('\n=== Districts with 0 destinations ===');
districts.forEach(dist => {
    if (dist.destinationCount === 0) {
        console.log(`  ${dist.name} (${dist.division})`);
    }
});

console.log('\n=== Total destinations:', destinations.length);
console.log('Total districts:', districts.length);

// Check for duplicates
const ids = destinations.map(d => d.id);
const duplicates = ids.filter((id, index) => ids.indexOf(id) !== index);
if (duplicates.length > 0) {
    console.log('\n=== DUPLICATE IDs:', duplicates, '===');
} else {
    console.log('\n=== No duplicate IDs ===');
}

// Check coordinates
const invalidCoords = [];
destinations.forEach(d => {
    const lat = d.latitude;
    const lng = d.longitude;
    if (lat === undefined || lat === null || lng === undefined || lng === null) {
        invalidCoords.push(`${d.id}: missing coords`);
    } else if (typeof lat !== 'number' || typeof lng !== 'number') {
        invalidCoords.push(`${d.id}: invalid type`);
    } else if (lat < -90 || lat > 90 || lng < -180 || lng > 180) {
        invalidCoords.push(`${d.id}: out of range ${lat},${lng}`);
    }
});

if (invalidCoords.length > 0) {
    console.log('\n=== INVALID COORDINATES:', invalidCoords, '===');
} else {
    console.log('\n=== All coordinates valid ===');
}

// Check required fields
const missingFields = [];
destinations.forEach(d => {
    ['id', 'name', 'district', 'division', 'category', 'latitude', 'longitude', 'shortDescription'].forEach(field => {
        if (!d[field]) missingFields.push(`${d.id}: missing ${field}`);
    });
});
if (missingFields.length > 0) {
    console.log('\n=== MISSING FIELDS:', missingFields, '===');
} else {
    console.log('\n=== All required fields present ===');
}

File: analyze_data.py
import json

with open('D:/Travel-Weather-Website/data/destinations.json', 'r') as f:
    destinations = json.load(f)

with open('D:/Travel-Weather-Website/data/districts.json', 'r') as f:
    districts = json.load(f)

# Analyze existing destinations by district
existing_by_district = {}
for d in destinations:
    dist = d.get('district')
    if dist not in existing_by_district:
        existing_by_district[dist] = []
    existing_by_district[dist].append(d['id'])

print("=== Existing destinations by district ===")
for dist, ids in sorted(existing_by_district.items()):
    print(f"{dist}: {len(ids)} - {ids}")

print("\n=== Districts with 0 destinations ===")
for dist in districts:
    if dist['destinationCount'] == 0:
        print(f"  {dist['name']} ({dist['division']})")

print("\n=== Total destinations:", len(destinations))
print("Total districts:", len(districts))

# Also check for duplicates
ids = [d['id'] for d in destinations]
duplicates = [id for id in set(ids) if ids.count(id) > 1]
if duplicates:
    print(f"\n=== DUPLICATE IDs: {duplicates} ===")
else:
    print("\n=== No duplicate IDs ===")

# Check coordinates
invalid_coords = []
for d in destinations:
    lat = d.get('latitude')
    lng = d.get('longitude')
    if lat is None or lng is None:
        invalid_coords.append(f"{d['id']}: missing coords")
    elif not isinstance(lat, (int, float)) or not isinstance(lng, (int, float)):
        invalid_coords.append(f"{d['id']}: invalid type")
    elif lat < -90 or lat > 90 or lng < -180 or lng > 180:
        invalid_coords.append(f"{d['id']}: out of range {lat},{lng}")

if invalid_coords:
    print(f"\n=== INVALID COORDINATES: {invalid_coords} ===")
else:
    print("\n=== All coordinates valid ===")